lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:js/js.dart';
+import 'package:js/js_util.dart' as js_util;
+
+@JS()
+@staticInterop
+class DomWindow {}
+
+extension DomWindowExtension on DomWindow {
+  external DomDocument get document;
+}
+
+@JS('window')
+external DomWindow get domWindow;
+
+@JS()
+@staticInterop
+class DomDocument {}
+
+extension DomDocumentExtension on DomDocument {
+  external List<DomNode> querySelectorAll(String selectors);
+}
+
+@JS()
+@staticInterop
+class DomEventTarget {}
+
+@JS()
+@staticInterop
+class DomNode extends DomEventTarget {}
+
+@JS()
+@staticInterop
+class DomHTMLElement extends DomNode {}
+
+@JS()
+@staticInterop
+class DomHTMLMetaElement {}
+
+extension DomHTMLMetaElementExtension on DomHTMLMetaElement {
+  external String get name;
+  external String get content;
+
+  external set name(String value);
+}
+
+Object domGetConstructor(String constructorName) =>
+    js_util.getProperty(domWindow, constructorName);
+
+extension DomHTMLListExtension on Iterable<DomNode> {
```

Comment:
Should `DomNode` be nullable?

Reply:
I don't think it's needed for your purposes (as `querySelectorAll` should only return non-null elements). Since we're being explicit here that it's for non-nullable iterables, it's fine to keep it non-nullable imo.

Reply:
I ended up just inlining this helper anyways. I grepped around and I didn't see another obvious use of this pattern. If we see this crop up again, we can generalize it at that point.